src/streamlink_cli/argparser.py

```diff
@@ -762,6 +762,21 @@ def build_parser():
         the ignored break.
         """
     )
+    transport.add_argument(
+        "--hls-segment-skip-discontinuity",
+        action="store_true",
+        help="""
+        Ignore segments in the playlist that occur during a discontinuity. This
+        option can be used to skip ads that are inserted in to a stream and marked
+        as a discontinuity.
+
+        Default is False.
+
+        Note: This option should be used with care, if the stream is running an ad
```

Comment:
@beardypig do you think it would make sense filtering plugin specific tags/attributes for detecting this case?
See https://github.com/streamlink/streamlink/issues/2357#issuecomment-473501203
Twitch live segments are annotated in the format `#EXTINF:2.000,live` and ad segments in the format `#EXTINF:2.002,87d22d0b208d6a47c3bba6f412f7fb43-03-04-2019-05.03.22`

Or are the metadata tags included at the beginning of all playlists during stream ads and not just the first one when the channel starts running ads?

Reply:
I didn't want it to be enabled by default in any plugin, but it could be. I didn't check, but I assumed that the format for VOD would be similar to the ads...

Reply:
Hm, yeah, maybe, I didn't think about that. :confused:

Reply:
Would appreciate this option — I only watch live streams and right now it's a crapshoot if I get an ad on launch.